Documentation/bpf.rst

```diff
@@ -550,16 +550,44 @@ maximum number of nesting calls respectively allowed call frames is ``8``.
 A caller can pass pointers (e.g. to the caller's stack frame) down to the
 callee, but never vice versa.
 
-BPF to BPF calls are currently incompatible with the use of BPF tail calls,
-since the latter requires to reuse the current stack setup as-is, whereas
-the former adds additional stack frames and thus changes the expected layout
-for tail calls.
-
 BPF JIT compilers emit separate images for each function body and later fix
 up the function call addresses in the image in a final JIT pass. This has
 proven to require minimal changes to the JITs in that they can treat BPF to
 BPF calls as conventional BPF helper calls.
 
+Up to kernel 5.9, BPF tail calls and BPF subprograms excluded each other. BPF
+programs that utilized tail calls couldn't take the benefit of shrinked program
```

Comment:
nit: `benefit of shrinked program` -> `benefit of reducing program image size`

Reply:
Please also run `./Documentation/update-spelling_wordlist.sh func unwinded` so the latter two get added to the word list and don't cause doc CI failure. Thanks for extending the doc!